WordPress/openverse
# Focus the trigger when content setting modal is closed

## Fixes



Fixes #4337 by @obulat

## Description


This PR adds the `triggerElement` prop to the `VInputModal`'s modal content to focus the trigger when modal is hidden.
In addition, it also fixes the search bar button: when they are focused, only the inner focus ring is visible instead of the current broken two rings:

<img width="637" alt="Screenshot 2024-05-15 at 8 14 15 PM" src="https://github.com/WordPress/openverse/assets/15233243/068f576a-3264-4eb9-9224-e430075bdfbf">

## Testing Instructions


1. Go to http://localhost:8443/search?q=cat, and narrow your screen to mobile width.
2. Click on the content settings button inside the search bar.
3. Press <kbd>Tab</kbd>
4. The content setting button should be focused (with a pink ring around it).
5. Compare this with the production: on openverse.org/search?q=cat, follow the same steps. Nothing should look focused.  If you press <kbd>Tab</kbd> again, you'll see that "Skip to content" button is focused, which shows that the body was focused when the modal was hidden.

## Checklist


      PRs) or the media properties generator (`just catalog/generate-docs media-props`
      for the catalog or `just api/generate-docs` for the API) where applicable.

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
